apps/condo/domains/miniapp/schema/B2BAppPromoBlock.test.js

```diff
@@ -0,0 +1,134 @@
+            })
+            test('Support can', async () => {
+                const subtitle = faker.commerce.productName()
+                expect(block).toHaveProperty('subtitle')
+                expect(block.title).not.toEqual(subtitle)
+                const [updated] = await updateTestB2BAppPromoBlock(support, block.id, {
+                    subtitle,
+                })
+                expect(updated).toHaveProperty('subtitle', subtitle)
+            })
+            test('User cannot', async () => {
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await updateTestB2BAppPromoBlock(user, block.id, {
+                        title: 'My title',
+                    })
+                })
+            })
+            test('Anonymous cannot', async () => {
+                await expectToThrowAuthenticationErrorToObj(async () => {
+                    await updateTestB2BAppPromoBlock(anonymous, block.id, {
+                        title: 'My title',
+                    })
+                })
+            })
+        })
+        describe('Read', () => {
+            let block
+            beforeAll(async () => {
+                [block] = await createTestB2BAppPromoBlock(admin)
+            })
+            const canReadRoles = [['admin'], ['support'], ['user']]
+            test.each(canReadRoles)('%p can read all', async (role) => {
+                const actor = actors[role]
+                const blocks = await B2BAppPromoBlock.getAll(actor, { id: block.id })
+                expect(blocks).toHaveLength(1)
+                expect(blocks[0]).toHaveProperty('id')
+                expect(blocks[0]).toHaveProperty('title')
+                expect(blocks[0]).toHaveProperty('subtitle')
+                expect(blocks[0]).toHaveProperty('textVariant')
+                expect(blocks[0]).toHaveProperty('backgroundColor')
+                expect(blocks[0]).toHaveProperty(['backgroundImage', 'publicUrl'])
+                expect(blocks[0]).toHaveProperty('url')
+                expect(blocks[0]).toHaveProperty('priority')
+            })
+            test('Anonymous cannot read', async () => {
+                await expectToThrowAuthenticationErrorToObjects(async () => {
+                    await B2BAppPromoBlock.getAll(anonymous, { id: block.id })
+                })
+            })
+        })
+        describe('Hard-delete', () => {
+            let block
+            beforeAll(async () => {
+                [block] = await createTestB2BAppPromoBlock(admin)
+            })
+            const noDeleteRoles = [['admin'], ['support'], ['user'], ['anonymous']]
+            test.each(noDeleteRoles)('%p cannot hard-delete', async (role) => {
+                const actor = actors[role]
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await B2BAppPromoBlock.delete(actor, block.id)
+                })
+            })
+        })
+    })
```

Comment:
What about the test for `backgroundColor` validation?

Reply:
Added some tests for that